ui/src/selection-panel/selection-panel-geometry-section.tsx

```diff
@@ -169,24 +180,54 @@ const SelectionPanelGeometrySection: React.FC<GeometryPlansPanelProps> = ({
         planIdsInViewport.some((planId) => planId === p.id),
     );
 
+    function fetchPlansAndSetVisible(planIds: GeometryPlanId[]) {
+        fetchPlanLayouts(planIds).then((plans) =>
+            plans
+                .filter(filterNotEmpty)
+                .forEach((plan) => onTogglePlanVisibility(wholePlanVisibility(plan))),
+        );
+    }
+
     const toggleAllPlanVisibilities = () => {
         if (visiblePlansInView.length > 0) {
             visiblePlansInView.forEach(onTogglePlanVisibility);
         } else if (planHeadersDisplayableInPanel.length === planHeaderCount) {
-            fetchPlanLayouts(planHeadersDisplayableInPanel.map((h) => h.id)).then((plans) =>
-                plans
-                    .filter(filterNotEmpty)
-                    .forEach((plan) => onTogglePlanVisibility(wholePlanVisibility(plan))),
-            );
+            const allDisplayablePlanIds = planHeadersDisplayableInPanel.map((h) => h.id);
+            fetchPlansAndSetVisible(allDisplayablePlanIds);
         }
     };
 
+    const visibleProjectIds = planHeadersDisplayableInPanel
+        .filter((plan) => visiblePlans.some((visiblePlan) => visiblePlan.id == plan.id))
```

Comment:
En merkkaa näitä kaikkia erikseen, mutta täällä tiedostossa näkyi aika paljon tuplayhtäsuuruuksia. Nämä voisi vaihtaa triploiksi